drivers/wifi/eswifi/eswifi_core.c

```diff
@@ -120,6 +120,27 @@ static void __parse_scan_res(char *str, struct wifi_scan_result *res)
 	}
 }
 
+char *eswifi_rsp_to_data(char *resp, size_t rlen)
+{
+	char *data = NULL;
+	int i;
+
+	/* format is \r\n[DATA]\r\nOK\r\n> or */
+	if (resp[0] != '\r' || resp[1] != '\n') {
+		return NULL;
+	}
+
+	for (i = 2; i < rlen; i++) {
+		if (!strncmp(&resp[i], "\r\nOK\r\n>", 7)) {
```

Comment:
What if "\r\nOK\r\n>" is embedded in [DATA]?

Reply:
well, this is a problem, since the "\r\nOK\r\n>" string would be incorrectly discarded.  I don't see any obvious solution in that case. I can contact Inventek about this.

ref: https://www.inventeksys.com/iwin/getting-started-guide/

Reply:
> well, this is a problem, since the "\r\nOK\r\n>" string would be incorrectly discarded

That's why I left comment https://github.com/zephyrproject-rtos/zephyr/pull/12270#discussion_r245250617 asking to add a code comment quickly describing the way data is encoded (e.g. quoted). Because it's impossible to transfer arbitrary data in another protocol (AT in this case) without encapsulating/encoding it. And question how that's done is one of the primary questions to be aware of when dealing with hardware like that. 

(And why I don't like hardware with AT protocol, because oftentimes these matters are described in the dark corners of manuals or not described at all. E.g. ESP8266 when it just popped up, had a lot of issues with its AT firmware. I'm happy to hear they're resolved by now, seeing https://github.com/zephyrproject-rtos/zephyr/pull/10539 - I for one never used ESP8266 with AT firmware "for real" at all.)

Reply:
Would it work if you looked for that particular string from the end of the received data? Maybe using a variant of `strrstr()` for arbitrary data (e.g. `memrmem()`).

Reply:
Also, the code, as written, might overflow `resp` if it reaches the the end of `resp` without finding the terminator string.

Reply:
> Would it work if you looked for that particular string from the end of the received data? Maybe using a variant of `strrstr()` for arbitrary data (e.g. `memrmem()`).

Yes, that would work, I need to rework a bit the spi code though since in current state this would request zeroing the quite large receiving buffer before spi recv.